modules/apps/27-interchain-accounts/host/keeper/handshake.go

```diff
@@ -47,8 +48,20 @@ func (k Keeper) OnChanOpenTry(
 		return "", err
 	}
 
-	if activeChannelID, found := k.GetOpenActiveChannel(ctx, connectionHops[0], counterparty.PortId); found {
-		return "", sdkerrors.Wrapf(icatypes.ErrActiveChannelAlreadySet, "existing active channel %s for portID %s", activeChannelID, portID)
+	activeChannelID, found := k.GetActiveChannelID(ctx, connectionHops[0], counterparty.PortId)
```

Comment:
If this whole block of code is the same as the one in `OnChannelOpenInit`, could we extract it into a helper function?

Reply:
I think it is fine to leave as is/be explicit. There's still a lot of duplicate code between OpenInit and OpenTry